indra/llcommon/lluuid.cpp

```diff
@@ -400,11 +401,16 @@ LLUUID LLUUID::operator^(const LLUUID& rhs) const
 	return id;
 }
 
+// WARNING: this algorithm SHALL NOT be changed. It is also used by the server
+// and plays a role in some assets validation (e.g. clothing items). Changing
+// it would cause invalid assets.
 void LLUUID::combine(const LLUUID& other, LLUUID& result) const
 {
-	HBXXH128 hash((const void*)mData, 16, false);	// false = do not finalize
-	hash.update((const void*)other.mData, 16);
-	hash.digest(result);
+	LLMD5 md5_uuid;
+	md5_uuid.update((unsigned char*)mData, 16);
+	md5_uuid.update((unsigned char*)other.mData, 16);
+	md5_uuid.finalize();
+	md5_uuid.raw_digest(result.mData);
```

Comment:
Won't it be better to revert whole lluuid class? De we use uuid hashing often per second?

Reply:
Nah, the rest is safe (random UUID creation is safe, since... random).